R/05c-rural-analysis.R

```diff
@@ -0,0 +1,364 @@
+    n_rural_popcenter = sum(urban_rural_statscan_popcenter == "RURAL", na.rm = TRUE),
+    n_missing_statscan_fsa = sum(is.na(urban_rural_statscan_fsa), na.rm = TRUE),
+    p_rural_statscan_fsa = 100*sum(urban_rural_statscan_fsa == "RURAL", na.rm = TRUE) / n(),
+    p_rural_popcenter = 100*sum(urban_rural_statscan_popcenter == "RURAL", na.rm = TRUE) / n()
+  ) |>
+  pivot_longer(
+    cols = starts_with(c("n_", "p_")),
+    names_to = "method",
+    values_to = "count"
+)
+
+rural_summary_by_method |> write_csv(
+  glue("{TABLES_OUT}/rural-summary-by-method.csv")
+)
+
+
+# =========================================================================== #
+# Roll up of rural flag to catchment ----
+# =========================================================================== #
+# --- Create a summary table of rural/urban classification by catchment,
+# --- according to # of addresses assigned
+catchment_rural_summary <- residence_region_crosswalk |>
+  st_drop_geometry() |>
+  left_join(complete_assignments, by = "dbid") |>
+  group_by(assigned) |>
+  summarise(
+    n_residences = n(),
+    n_rural_statscan_fsa = sum(urban_rural_statscan_fsa == "RURAL", na.rm = TRUE),
+    n_rural_popcenter = sum(urban_rural_statscan_popcenter == "RURAL", na.rm = TRUE),
+    p_rural_statscan_fsa = 100 * sum(urban_rural_statscan_fsa == "RURAL", na.rm = TRUE) / n(),
+    p_rural_popcenter = 100 * sum(urban_rural_statscan_popcenter == "RURAL", na.rm = TRUE) / n(),
+    is_rural_statscan_fsa = if_else(p_rural_statscan_fsa > 50, "RURAL", "URBAN"),
+    is_rural_popcenter = if_else(p_rural_popcenter > 50, "RURAL", "URBAN"),
+    .groups = 'drop'
+  )
+
+catchment_rural_summary
+
+# note that due to an abundance of rural addresses, there are many more rural catchments than I might have expected
+# does this mean that using address is misleading, and we should use population? what would that look like?
+# or are they truly placed out in rural areas? 
+catchment_rural_summary |> 
+  summarize(
+    mean_rural_statscan_fsa = mean(p_rural_statscan_fsa, na.rm = TRUE),
+    mean_rural_popcenter = mean(p_rural_popcenter, na.rm = TRUE),
+    median_rural_statscan_fsa = median(p_rural_statscan_fsa, na.rm = TRUE),
+    median_rural_popcenter = median(p_rural_popcenter, na.rm = TRUE)
+  ) |>
+  pivot_longer(everything())
+
+catchment_rural_summary |>
+  count(is_rural_statscan_fsa, is_rural_popcenter) |>
+  arrange(desc(n))
+
+# =========================================================================== #
+# look at similar results with population estimates
+# =========================================================================== #
+db_pop_est <- db_projections_transformed_agg  |>
+  select(dbid, population, geometry = geom, csdid)
+
+# generate a crosswalk that maps each residence to a region, for each boundary (method).
+popcenter_dbs_crosswalk_statscan <- resides_in_region(db_pop_est, pop_centers, "dbid", "pcname")
```

Comment:
I wonder if this method is too strict for comparing db regions to the popcenter/fsa regions? I.e. if a DB ever straddles a region, it's going to be 'false' for every single possible region, which seems incorrect. A different method might be to 'vote' a DB to be rural/urban based on the addresses within first (similar to how we assigned DBs to catchments), and then do the roll ups after that? (Although that won't fix all the dbs with no addresses). 

Another option might be to use a less restrictive st function than `st_within` - e.g. `st_intersection` computes the area overlap between two polygons, then you can compute what proportion of the DB lies within each region, and assign it to the largest one?